core/src/main/java/io/aiven/klaw/service/CommonUtilsService.java

```diff
@@ -358,34 +368,53 @@ public String getBaseUrl() {
           + kwContextPath;
   }
 
+  public Map<String, String> getBaseIpUrlFromEnvironment() {
+    if (baseUrlsMap != null && !baseUrlsMap.isEmpty()) {
+      return baseUrlsMap;
+    } else {
+      baseUrlsMap = new HashMap<>();
+      String hostAddress = InetAddress.getLoopbackAddress().getHostAddress();
+      String hostName = InetAddress.getLoopbackAddress().getHostName();
+      int port = Integer.parseInt(Objects.requireNonNull(environment.getProperty("server.port")));
+
+      baseUrlsMap.put(BASE_URL_ADDRESS, hostAddress + ":" + port);
+      baseUrlsMap.put(BASE_URL_NAME, hostName + ":" + port);
+      return baseUrlsMap;
+    }
+  }
+
   public void resetCacheOnOtherServers(KwMetadataUpdates kwMetadataUpdates) {
     log.info("invokeResetEndpoints");
     try {
-      String tenantName =
-          manageDatabase
-              .getHandleDbRequests()
-              .getMyTenants(kwMetadataUpdates.getTenantId())
-              .get()
-              .getTenantName();
-
-      // if property is null refresh memory directly
       if (uiApiServers != null && uiApiServers.length() > 0) {
-        updateMetadata(kwMetadataUpdates);
-      } else {
-        String[] servers = new String[0];
-        if (uiApiServers != null) {
-          servers = uiApiServers.split(",");
-        }
-
+        String[] servers = uiApiServers.split(",");
         String basePath;
         for (String server : servers) {
 
-          if ("".equals(kwContextPath)) basePath = server;
-          else basePath = server + "/" + kwContextPath;
+          if ("".equals(kwContextPath)) {
+            basePath = server;
+          } else {
+            basePath = server + "/" + kwContextPath;
+          }
+
+          Map<String, String> baseUrlsFromEnv = getBaseIpUrlFromEnvironment();
+
+          // ignore metadata cache reset on local.
+          if (baseUrlsFromEnv != null && !baseUrlsFromEnv.isEmpty()) {
+            if (baseUrlsFromEnv.containsKey(BASE_URL_ADDRESS)
+                && basePath.contains(baseUrlsFromEnv.get(BASE_URL_ADDRESS))) {
+              continue;
+            }
+            if (baseUrlsFromEnv.containsKey(BASE_URL_NAME)
+                && basePath.contains(baseUrlsFromEnv.get(BASE_URL_NAME))) {
+              continue;
```

Comment:
Would it be easier to remove the base url from the uiApiServes string?

Reply:
Was thinking it's easy to copy the same string across all servers, and code will handle it.

Reply:
I mean on startup once we build that url we can remove it from uiApiServers and then that loop with the checks doesnt happen each time the request comes in, so the same string can be used across all servers, but then we manage it at startup by removing it from the String.